instrumentation/servlet/src/main/java/brave/servlet/TracingFilter.java

```diff
@@ -0,0 +1,89 @@
+package brave.servlet;
+
+import brave.Span;
+import brave.Tracer;
+import brave.Tracing;
+import brave.http.HttpServerParser;
+import brave.propagation.TraceContext;
+import java.io.IOException;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public final class TracingFilter implements Filter {
```

Comment:
@johnament FYI there will be some adjustments tomorrow (likely HttpTracing, which will be a component that includes Tracing and default http client and server tagging policy). However, most of this code will look the same. We won't be adding new Brave 3 style instrumentation moving forward. If you want feign or CDI, it would be based on the new stuff. Have a look!